code/game/objects/structures.dm

```diff
@@ -82,42 +82,34 @@
 		return FALSE
 	var/blocking_object = density_check()
 	if(blocking_object)
-		to_chat(user, "<span class='warning'>You cannot climb [src], as it is blocked by \a [blocking_object]!</span>")
+		to_chat(user, "<span class='warning'>You cannot climb onto [src], as it is blocked by \a [blocking_object]!</span>")
 		return FALSE
 
-	var/turf/T = src.loc
-	if(!T || !istype(T)) return FALSE
+	if(!isturf(loc))
+		return FALSE
 
-	climber = user
-	if(HAS_TRAIT(climber, TRAIT_TABLE_LEAP))
+	if(HAS_TRAIT(user, TRAIT_TABLE_LEAP))
 		user.visible_message("<span class='warning'>[user] gets ready to vault up onto [src]!</span>")
 		if(!do_after(user, 0.5 SECONDS, target = src))
-			climber = null
 			return FALSE
 	else
 		user.visible_message("<span class='warning'>[user] starts climbing onto [src]!</span>")
 		if(!do_after(user, 5 SECONDS, target = src))
-			climber = null
 			return FALSE
 
 	if(!can_touch(user) || !climbable)
-		climber = null
 		return FALSE
 
-	var/old_loc = usr.loc
-	user.loc = get_turf(src)
-	user.Moved(old_loc, get_dir(old_loc, usr.loc), FALSE)
+	var/old_loc = user.loc
+	user.forceMove(get_turf(src))
+	user.Moved(old_loc, get_dir(old_loc, user.loc), FALSE)
```

Comment:
is this one still needed due to the use of forceMove?

Reply:
probably not